ipam.go

```diff
@@ -44,6 +44,10 @@ type Ipamer interface {
 	Load(ctx context.Context, dump string) error
 	// ReadAllPrefixCidrs retrieves all existing Prefix CIDRs from the underlying storage
 	ReadAllPrefixCidrs(ctx context.Context) ([]string, error)
+	// CreateNamespace creates a namespace for overlapping prefixes
+	CreateNamespace(ctx context.Context, namespace string) error
+	// DeleteNamespace deletes a namespace for overlapping prefixes
+	DeleteNamespace(ctx context.Context, namespace string) error
```

Comment:
Hate to say, but the caller has no chance to see if the namespace is already there, he must check for an error during NewPrefix, what about adding a either `NameSpaceExists` or CRUD to Namespace ?

Reply:
Yeah this needs a bit more thought. Another thing that's bugging me is that these methods take namespace as an argument while the others use a context 🤔 

I think the best option is for each `Storage` to handle namespace creation and not to expose that directly to a user - like in my original PR... but with some better checks to ensure the namespace exists.

If we need an API to drop tables used for namespaces in the SQL storage, that should be `ipamer.DeleteAllPrefixes(context.Context)` where the most efficient implementation would be `DROP TABLE IF EXISTS prefixes_tenant-a` - and in the case of `namespace == root` we'd just drop all rows in that table. Storage already has an API for this, it's just not exposed via ipamer.

There were also other issues that arose from `CreateNamespace`/`DeleteNamespace` that I hadn't thought of.
It's possible to call Create/DeleteNamespace across different threads, and for you to end up with odd behaviour (i.e CreateNamespace, CreatePrefix in one thread... but another thread called DeleteNamespace between these two events... and you get SQL errors from CreatePrefix). Letting Storage handle namespaces would solve the problem also.

Reply:
What about keeping the approach you had before and just add List/Get/Delete Namespace ? Just thinking.